pkg/v2/tkr/controller/tkr-status/clusterstatus/clusterstatus.go

```diff
@@ -175,32 +177,45 @@
 		vLabelMinor(major, minor),
 		vLabelMinor(major, minor+1),
 	} {
-		updateVersion, err := r.findUpdateVersion(tkr, cluster, clusterClass, versionPrefix)
+		updateVersions, err := r.findUpdateVersion(tkr, cluster, clusterClass, versionPrefix)
 		if err != nil {
 			return nil, err
 		}
-		if updateVersion != "" {
-			result = append(result, updateVersion)
-		}
+		result = append(result, updateVersions...)
 	}
 
 	return result, nil
 }
 
-func (r *Reconciler) findUpdateVersion(tkr *runv1.TanzuKubernetesRelease, cluster *clusterv1.Cluster, clusterClass *clusterv1.ClusterClass, versionPrefix string) (string, error) {
+func (r *Reconciler) findUpdateVersion(tkr *runv1.TanzuKubernetesRelease, cluster *clusterv1.Cluster, clusterClass *clusterv1.ClusterClass, versionPrefix string) ([]string, error) {
 	query, err := resolution.ConstructQuery(versionPrefix, cluster, clusterClass)
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 	tkrResult := r.TKRResolver.Resolve(*query)
 	if tkrResult.ControlPlane.K8sVersion != "" && tkrResult.ControlPlane.TKRName != tkr.Name {
-		if tkrResult.ControlPlane.K8sVersion == cluster.Spec.Topology.Version {
-			resolvedTKR := tkrResult.ControlPlane.TKRsByK8sVersion[tkrResult.ControlPlane.K8sVersion][tkrResult.ControlPlane.TKRName]
-			return resolvedTKR.Spec.Version, nil
+		return resolvedTKRVersions(tkr, tkrResult)
+	}
+	return nil, nil
+}
+
+func resolvedTKRVersions(currentTKR *runv1.TanzuKubernetesRelease, tkrResult data.Result) ([]string, error) {
+	var result []string
+	for _, tkrs := range tkrResult.ControlPlane.TKRsByK8sVersion {
+		for _, tkr := range tkrs {
+			tkrVersion, _ := version.ParseSemantic(tkr.Spec.Version)
+			currentTKRVersion, _ := version.ParseSemantic(currentTKR.Spec.Version)
```

Comment:
nit: Should we consider moving this statement out of the loop: It is not going to change.

Reply:
Done!